redis/reflect_go117.go

```diff
@@ -0,0 +1,34 @@
+//go:build go1.17 && !go1.18
+// +build go1.17,!go1.18
```

Comment:
This will cause builds older than go 1.17 to fail
```
➜  redigo git:(fix-anonymous-nil) ✗ go version
go version go1.16 darwin/arm64
➜  redigo git:(fix-anonymous-nil) ✗ go build ./...
# github.com/gomodule/redigo/redis
redis/scan.go:654:14: undefined: fieldByIndexErr
```

I _think_ based on [this](https://stackoverflow.com/a/38439941) that !go1.17 will refer to everything older than 1.18. This allowed go1.16 to build and pass tests, but earlier versions of golang are not supported on my m1 so i couldn't test
```suggestion
//go:build !go1.18
// +build !go1.18
```